metadata-ingestion/src/datahub/ingestion/source/looker_common.py

```diff
@@ -515,23 +515,28 @@ def from_api(  # noqa: C901
                 else:
                     aliased_explore = False
```

Comment:
in the case where the explore is an unaliased explore and contains a single join without using dependent fields, we should also add the same named view as an upstream dependency. This means adding a:
```
views.add(explore.name)
```
in this `else` section?

Reply:
You should be able to validate this by adding another mock test explore that behaves that way.

Reply:
:thinking: :thinking: 

I'm finding that entire area of code confusing, especially where it attempts to change the view name from the dependent fields where it may also be the name of the explore and the explore is aliased.

I'm going to add some comments in there to try to clarify what's happening and why

Reply:
yup this looks great!